lib/puppet/application/device.rb

```diff
@@ -170,10 +180,17 @@ def main
     returns = Puppet.override(:current_environment => env, :loaders => Puppet::Pops::Loaders.new(env)) do
       # find device list
       require 'puppet/util/network_device/config'
-      devices = Puppet::Util::NetworkDevice::Config.devices
+      devices = Puppet::Util::NetworkDevice::Config.devices.dup
+      if options[:target]
+        devices.select! { |key, value| key == options[:target] }
+      end
       if devices.empty?
-        Puppet.err "No device found in #{Puppet[:deviceconfig]}"
-        exit(1)
+        if options[:target]
+          Puppet.err "Target device/certificate '#{options[:target]}' not found in #{Puppet[:deviceconfig]}"
```

Comment:
Please mark this string for externalization. See https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md#writing-translatable-code.